### PR TITLE
Feature/24 - Hosted By

### LINKED ISSUE #24: Ability to describe the host for software
Should be possible to describe software which will run under the same process or in the same container or on the same machine.

Later it will be possible to describe grouping by network configuration but that is not part of this task.

### PR DESCRIPTION
**Pull request by Sub-Xaero**
Closes #24 

Work in progress.

### CODE REVIEW

File: acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/HostEntity.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.ephyra.acropolis.persistence.api.entity
+
+import javax.persistence.*
+
+@Entity
+class HostEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    var id: Int = 0
```
[the remark this answers is not in the document]
Comment (Sub-Xaero) [reply acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/HostEntity.kt]: Sure. Copied from `ProjectEntity`, so should that be `Long?` too?

File: acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/impl/HostService.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class HostService : IHostService {
+    @Autowired
+    private lateinit var persistence: HostPersistence
+
+    override fun createHost(name: String, hostedBy: String) {
+        println("Acropolis service is creating host with name $name")
```
[the remark this answers is not in the document]
Comment (Sub-Xaero) [reply acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/impl/HostService.kt]: Copied from every other service. My `Ctrl + C, Ctrl + V` skills are strong

**Comment by ThetaSinner** [comment]
What's the status of this? :)

File: acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt [hunk 3]
```diff
@@ -16,15 +16,15 @@ import javax.persistence.*
  *  - An interactive shell or terminal
  */
 @Entity
-data class ApplicationSoftwareEntity @JvmOverloads constructor (
-    @Column(nullable=false)
-    var name: String,
+data class ApplicationSoftwareEntity @JvmOverloads constructor(
+        @Column(nullable = false)
```
Comment (ThetaSinner) [inline acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt]: I swear there is a bug in the formatter for Kotlin. It sometimes leaves these and sometimes indents them more. Oh well, doesn't make much difference :)

File: acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/api/IDatastoreService.kt [hunk 4]
```diff
@@ -3,6 +3,6 @@ package org.ephyra.acropolis.service.api
 import org.ephyra.acropolis.persistence.api.entity.DatastoreEntity
 
 interface IDatastoreService {
-    fun create(projectId: Long, name: String)
+    fun create(projectId : Long, name: String)
```
Comment (ThetaSinner) [inline acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/api/IDatastoreService.kt]: extra space before :

File: acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/api/ISystemSoftwareService.kt [hunk 5]
```diff
@@ -4,6 +4,5 @@ import org.ephyra.acropolis.persistence.api.entity.SystemSoftwareEntity
 
 interface ISystemSoftwareService {
     fun create(projectId : Long, name: String)
```
Comment (ThetaSinner) [inline acropolis-service/src/main/kotlin/org/ephyra/acropolis/service/api/ISystemSoftwareService.kt]: at some point these will need doc comments on them which is why I left a gap - to cause myself guilt :P

File: acropolis-shell/src/main/kotlin/org/ephyra/acropolis/shell/CreateCommand.kt [hunk 6]
```diff
@@ -40,8 +40,7 @@ class CreateCommand {
         val project = appState.currentProject
         if (project != null) {
             systemSoftwareService.create(project.id, name)
-        }
-        else {
+        } else {
```
Comment (ThetaSinner) [inline acropolis-shell/src/main/kotlin/org/ephyra/acropolis/shell/CreateCommand.kt]: yuck

Comment (ThetaSinner) [reply acropolis-shell/src/main/kotlin/org/ephyra/acropolis/shell/CreateCommand.kt]: is that the default formatter? you haven't modified your rules?

If it is the default then let's just go with it

Comment (Sub-Xaero) [reply acropolis-shell/src/main/kotlin/org/ephyra/acropolis/shell/CreateCommand.kt]: Tis default indeed. And my favourite <3

File: acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt (hunk 3 above)
Comment (Sub-Xaero) [reply acropolis-persistence/src/main/kotlin/org/ephyra/acropolis/persistence/api/entity/ApplicationSoftwareEntity.kt]: Aye , Intellij always does it for me. No use fighting the inevitable

**Pull request merged**